Tutorials/WinForms/Classic/implement-crud-functionality-with-xpo-objects.md

```diff
@@ -0,0 +1,163 @@
+# Implement CRUD functionality with XPO objects
+
+[Step 2 <<](/connect-data-grid-to-xpo-objects.md) 
+&nbsp;&nbsp;&nbsp;
+[Back to TOC](../../)
+&nbsp;&nbsp;&nbsp;
+[>> Step 4](/bind-the-data-grid-to-large-data-source.md)   
+
+## Edit an object in a separate window and save changes
+* Create the **Forms** folder and invoke the context menu for it in the **Solution Explorer**. 
+* Click the **Add DevExpress Item**  to open the **DevExpress Template Gallery** window. 
+* In the **WinForms** tab page, select the **WinForms Common > Form** item and change the **Item Name** to **EditCustomerForm**. 
+* Click the **Add Item** button to open a new Form in a visual designer.
+*  Select the `Text` property in the **Properties** window and change it to **Edit Customer**.
+* Move the *Form1.cs* file to the *Forms* folder and rename the file to *CustomersListForm.cs*. Visual Studio should prompt whether to change the class name as well. If not, open the code editor and use the **Rename** menu command (**Ctrl+R, Ctrl+R**) to change the class name . Change the base class to `XtraForm`, open the designer, and set the `Text` property to **Customers**.
+* Drop the `XPBindingSource` and `DataLayoutControl` components from the toolbox and change their names to **CustomerBindingSource** and **CustomerLayoutControl**.
+* Rebuild the project.
+* Set the `CustomerBindingSource.ObjectClassInfo` property to **DxSample.DataAccess.Customer** *(select a value from the drop-down list and rebuild the project)*.
+* Set the `CustomerBindingSource.DisplayableProperties` property to **FirstName;LastName**.
+* Set the `CustomerLayoutControl.Dock` property to **Fill**.
+* Select the `CustomerLayoutControl` component on the design surface.
+* Click the **Retrieve Fields** smart tag command to open the **Select Binding Source** wizard (if the smart tag icon is not visible, press the **Esc** key to navigate from the root `LayoutGroup` to the `LayoutControl`).
```

Comment:
```suggestion
* Click the **Retrieve Field**'s smart tag command to open the **Select Binding Source** wizard (if the smart tag icon is not visible, press **Esc** to navigate from the root `LayoutGroup` to the `LayoutControl`).
```

Reply:
Why **Retrieve Fields**? You cannot use '**Retrieve Fields** smart tag command' when you are referring to something that belongs to something else - in English you should use a possessive.

Reply:
The "Retrieve Fields" is the command caption. That is why I did not use possessive. I have significantly changed this text (now I use "Select the **Retrieve Fields** item").

0b819ffccfc4f5db9c61aab87ceda368661f7e96